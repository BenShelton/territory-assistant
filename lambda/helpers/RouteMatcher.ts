import { APIGatewayEvent } from 'aws-lambda'

type APIMethod = 'GET' | 'POST' | 'PUT' | 'DELETE'

export class RouteMatcher {
  private _paramRegex = /:(.*?)(?=(\/|$))/
  private _event: APIGatewayEvent
  private _pathParams: Record<string, string> = {}

  constructor (event: APIGatewayEvent) {
    this._event = event
  }

  get pathParams (): Record<string, string> {
    return this._pathParams
  }

  private _getMatches (path: string, regex: RegExp): string[] {
    const matches = path.match(regex)
    if (!matches) return []
    return matches.slice(1)
  }

  testRoute (method: APIMethod, path: string): boolean {
    if (this._event.httpMethod !== method) return false
    const stringParamRegex = new RegExp(path.replace(this._paramRegex, '(\\w+)') + '/?$')
    if (!stringParamRegex.test(this._event.path)) return false
    const paramNames = this._getMatches(path, this._paramRegex)
    const paramValues = this._getMatches(this._event.path, stringParamRegex)
    this._pathParams = paramNames.reduce((acc: Record<string, string>, p, i) => {
      if (p) acc[p] = paramValues[i]
      return acc
    }, {})
    return true
  }
}
