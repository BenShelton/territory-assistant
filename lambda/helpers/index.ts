import { APIGatewayEvent } from 'aws-lambda'

type APIMethod = 'GET' | 'POST'
type RouteMatcher = (method: APIMethod, path: string) => boolean
interface IResult {
  statusCode: number
  body: string
}

export function generateRouteMatcher (event: APIGatewayEvent): RouteMatcher {
  return function routeMatcher (method: APIMethod, path: string) {
    if (event.httpMethod !== method) return false
    return new RegExp(path + '/?$').test(event.path)
  }
}

export function success<T> (data: T): IResult {
  return { statusCode: 200, body: JSON.stringify(data) }
}

export function badRequest (message: string): IResult {
  return { statusCode: 400, body: message }
}

export function notFound (): IResult {
  return { statusCode: 404, body: 'Unknown Route' }
}
