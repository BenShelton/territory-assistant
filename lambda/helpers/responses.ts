interface IResult {
  statusCode: number
  body: string
}

export function success<T> (data: T): IResult {
  return { statusCode: 200, body: JSON.stringify(data) }
}

export function badRequest (message: string): IResult {
  return { statusCode: 400, body: message }
}

export function unauthorized (): IResult {
  return { statusCode: 401, body: 'Unauthorized' }
}

export function notFound (): IResult {
  return { statusCode: 404, body: 'Unknown Route' }
}
