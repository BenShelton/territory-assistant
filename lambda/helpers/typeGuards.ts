import { IPoint } from 'types'

export function isObject (obj: unknown): obj is Record<string, unknown> {
  return typeof obj === 'object' && !Array.isArray(obj) && !!obj
}

export function isPointList (obj: unknown): obj is IPoint[] {
  if (!Array.isArray(obj)) return false
  const basePoint: IPoint = { lat: 0, lng: 0 }
  return Object.entries(basePoint)
    .every(([k, v]) => {
      return obj.every(p => {
        if (!isObject(p)) return false
        return typeof p[k] === typeof v
      })
    })
}
