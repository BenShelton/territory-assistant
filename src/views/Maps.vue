<template>
  <v-container>
    <h1 class="headline">
      Maps
    </h1>
  </v-container>
</template>

<script lang="ts">
import Vue from 'vue'
//

export default Vue.extend({
  name: 'Maps'
})
</script>
