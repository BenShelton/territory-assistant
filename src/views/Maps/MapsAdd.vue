<template>
  <v-container
    flex-column
    flex-nowrap
    align-start
    fill-height
  >
    <h1 class="headline mb-3">
      Add New Map
    </h1>
    <MapEditor @save="onSave" />
  </v-container>
</template>

<script lang="ts">
import Vue from 'vue'

import MapEditor from '@/components/MapEditor.vue'

import store from '@/store'

import { IMap } from 'types'

export default Vue.extend({
  name: 'MapsAdd',

  components: {
    MapEditor
  },

  methods: {
    onSave (map: IMap): void {
      store.dispatch.maps.add(map)
    }
  }
})
</script>
