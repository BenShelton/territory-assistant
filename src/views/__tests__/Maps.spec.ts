import { createWrapper } from 'UnitTest/helpers'

import Maps from '../Maps.vue'

describe('Views/Maps', () => {
  it('should have the name "Maps"', () => {
    const wrapper = createWrapper(Maps, { store: true })
    expect(wrapper.name()).toBe('Maps')
  })

  it('should match the snapshot', () => {
    const wrapper = createWrapper(Maps, { store: true })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
