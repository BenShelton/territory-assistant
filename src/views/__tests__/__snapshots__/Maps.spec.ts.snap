// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Views/Maps should match the snapshot 1`] = `
<div class="container flex-column flex-nowrap align-start fill-height">
  <h1 class="headline mb-3">
    Edit Maps
  </h1>
  <div class="editor v-card v-sheet theme--light">
    <!---->
    <div role="dialog" class="v-dialog__container">
      <div role="document" class="v-dialog__content" style="z-index: 0;">
        <div class="v-dialog" style="max-width: 300px; display: none;"></div>
      </div>
    </div>
    <div role="dialog" class="v-dialog__container">
      <div role="document" class="v-dialog__content" style="z-index: 0;">
        <div class="v-dialog" style="max-width: 300px; display: none;"></div>
      </div>
    </div> <button type="button" class="info-toggle-btn v-btn v-btn--contained theme--light v-size--default primary"><span class="v-btn__content">
    Toggle Labels
  </span></button>
    <div id="map"></div>
  </div>
</div>
`;
